CODE_STYLE.md

```diff
@@ -0,0 +1,3293 @@
+  end in a question mark.  (i.e. `Array#empty?`). Methods that don't return a
+  boolean, shouldn't end in a question mark.
+<sup>[[link](#bool-methods-qmark)]</sup>
+
+* <a name="dangerous-method-bang"></a>
+  The names of potentially *dangerous* methods (i.e. methods that modify
+  `self` or the arguments, `exit!` (doesn't run the finalizers like `exit`
+  does), etc.) should end with an exclamation mark if there exists a safe
+  version of that *dangerous* method.
+<sup>[[link](#dangerous-method-bang)]</sup>
+
+  ```Ruby
+  # bad - there is no matching 'safe' method
+  class Person
+    def update!
+    end
+  end
+
+  # good
+  class Person
+    def update
+    end
+  end
+
+  # good
+  class Person
+    def update!
+    end
+
+    def update
+    end
+  end
+  ```
+
+* <a name="safe-because-unsafe"></a>
+  Define the non-bang (safe) method in terms of the bang (dangerous) one if
+  possible.
+<sup>[[link](#safe-because-unsafe)]</sup>
+
+  ```Ruby
+  class Array
+    def flatten_once!
+      res = []
+
+      each do |e|
+        [*e].each { |f| res << f }
+      end
+
+      replace(res)
+    end
+
+    def flatten_once
+      dup.flatten_once!
+    end
+  end
+  ```
+
+* <a name="reduce-blocks"></a>
+  When using `reduce` with short blocks, name the arguments `|a, e|`
+  (accumulator, element).
+<sup>[[link](#reduce-blocks)]</sup>
+
+* <a name="other-arg"></a>
+  When defining binary operators, name the parameter `other`(`<<` and `[]` are
+  exceptions to the rule, since their semantics are different).
+<sup>[[link](#other-arg)]</sup>
+
+  ```Ruby
+  def +(other)
+    # body omitted
+  end
+  ```
+
+## Comments
+
+> Good code is its own best documentation. As you're about to add a
+> comment, ask yourself, "How can I improve the code so that this
+> comment isn't needed?" Improve the code and then document it to make
+> it even clearer. <br>
+> -- Steve McConnell
+
+* <a name="no-comments"></a>
+  Write self-documenting code and ignore the rest of this section. Seriously!
+<sup>[[link](#no-comments)]</sup>
+
+* <a name="english-comments"></a>
+  Write comments in English.
+<sup>[[link](#english-comments)]</sup>
+
+* <a name="hash-space"></a>
+  Use one space between the leading `#` character of the comment and the text
+  of the comment.
+<sup>[[link](#hash-space)]</sup>
+
+* <a name="english-syntax"></a>
+  Comments longer than a word are capitalized and use punctuation. Use [one
+  space](http://en.wikipedia.org/wiki/Sentence_spacing) after periods.
+<sup>[[link](#english-syntax)]</sup>
+
+* <a name="no-superfluous-comments"></a>
+  Avoid superfluous comments.
+<sup>[[link](#no-superfluous-comments)]</sup>
+
+  ```Ruby
+  # bad
+  counter += 1 # Increments counter by one.
+  ```
+
+* <a name="comment-upkeep"></a>
+  Keep existing comments up-to-date. An outdated comment is worse than no
+  comment at all.
+<sup>[[link](#comment-upkeep)]</sup>
+
+> Good code is like a good joke - it needs no explanation. <br>
```

Comment:
This quote is not good. Code can be good but missing immediate context and the decisions that produced the code. Not needing an explanation means all readers have the necessary context (background, etc) to understand the joke (or code).